### ISSUE TITLE
Time to live not set.

### ISSUE THREAD

**Issue by GohKianSeng** [redis/redis-om-spring#458]
hello this is my entity model.
set the time to live but in redis insight, show no limit.

<img width="620" alt="image" src="https://github.com/redis/redis-om-spring/assets/56799988/ae2e30a3-e63e-406e-8769-3f47d4ec9d6d">

<img width="708" alt="image" src="https://github.com/redis/redis-om-spring/assets/56799988/6686675c-ada8-4ab7-81a0-a31b04d9762c">

`@Document("TestResultRedis", timeToLive = 5 * 60)
data class TestResultRedisModel(
    @Id
    val id: Long?,

    @Searchable
    val uuid: String?,

    @Searchable
    val uatId: String?,

    @Searchable
    val msgType: String?,

    @Searchable
    val filename: String?,

    @Searchable
    val status: String?,

    @Searchable
    val result: String?,

    val overrideStatus: String?,

    val remarksSnapshotId: Long?,

    @Searchable
    val actualCodes: List<String>,

    val mappedActualCodes: List<String>,

    @Searchable
    val expectedCodes: List<String>,

    val mappedExpectedCodes: List<String>,

    @Searchable
    val expectedUserCodes: List<String>,

    val batchJobStatus: String?,

    val expectedBatchJobStatus: String?,

    val expectedUserBatchJobStatus: String?,

    @Searchable
    val remark: String?,

    val remarkDate: String?,

    val remarkUuid: String?,

    @Searchable
    val remarkHashTags: List<String>?,

    @Searchable
    val assignee: String?,

    val user: String?,

    @Searchable
    var kivIndicator: Boolean?,

    @Searchable
    val manualIndicator: Boolean?,

    val overwrite: Boolean?
)`

**Comment by GohKianSeng** [comment]
just realised if i used saveAll. the Time to live is set.
is that a bug?

<img width="355" alt="image" src="https://github.com/redis/redis-om-spring/assets/56799988/4abf05bc-133c-4c1a-9145-4e0f52dfb9ae">

**Comment by bsbodden** [comment]
Sounds like a bug!

**Comment by GohKianSeng** [comment]
yes i used repository.save

**Comment by GohKianSeng** [comment]
hello @bsbodden 

the controller save() has TTL set.
<img width="640" alt="image" src="https://github.com/redis/redis-om-spring/assets/56799988/2866b871-f94a-4052-a1cf-f09f179f9afc">

but using PostConstruct doesn't work
<img width="623" alt="image" src="https://github.com/redis/redis-om-spring/assets/56799988/cc55d653-9f64-475a-8d8f-5e99f7741e6d">

Redis Insight 
<img width="727" alt="image" src="https://github.com/redis/redis-om-spring/assets/56799988/2b1a4eb9-f6e5-4218-976b-b8a9bec05684">

**Comment by bsbodden** [comment]
@GohKianSeng I can't seem to be able to replicate the issue, could you create a simple reproducer app? Just models, repository and a CommandLineRunner to run your scenario?

**Comment by bsbodden** [comment]
@GohKianSeng I don't think we have added support for `@PostConstruct`, that might be the root of the issue here

**Comment by nexlink-giangbui** [comment]
hello @bsbodden, I also get the same issue, the documents is saved without TTL when using save()

**Comment by nexlink-giangbui** [comment]
hi @bsbodden, I found that in Springboot project, if we use spring-boot-devtools, then when Application is running, it will use class loader _org.springframework.boot.devtools.restart.classloader.RestartClassLoader_, not _jdk.internal.loader.ClassLoaders_  (using by spring-om), this  will lead to spring-om cannot load _KeyspaceConfiguration_ for entity at
 `com.redis.om.spring.RedisJSONKeyValueAdapter.getTTLForEntity(Object entity)`
because 
`entityClassKey = ClassLoader.getSystemClassLoader().loadClass(entity.getClass().getTypeName())`
will return entityClassKey that is not in _KeyspaceConfiguration.settingsMap_


In context of @SpringBootTest, the _jdk.internal.loader.ClassLoaders_ will be used, that why in this context the TTL is working but does not work in Application context with devtools

Thanks

**Comment by bsbodden** [comment]
@nexlink-giangbui thanks for the investigation! The RestartClassLoader getting in the way of low level stuff is a recurring theme, I'm looking into ways to get to the correct classloader in that scenario. Stay tuned!
